tests/crypto/src/common_test.c

```diff
@@ -0,0 +1,127 @@
+/*$$$LICENCE_NORDIC_STANDARD<2019>$$$*/
+
+#include <stdio.h>
+#include <stdarg.h>
+#include <string.h>
+#include <stdint.h>
+#include <stddef.h>
+#include <misc/printk.h>
+#include <drivers/entropy.h>
+#include <mbedtls/ctr_drbg.h>
+#include <sys/util.h>
+#include <sys_clock.h>
+
+#include "common_test.h"
+
+#include <mbedtls/memory_buffer_alloc.h>
+static unsigned char _mbedtls_heap[65536];
+#if defined(CONFIG_CC310_BACKEND)
+static mbedtls_platform_context platform_context = { 0 };
+#endif
+
+volatile uint32_t t_start;
+volatile uint32_t t_stop;
+
+int init_cc310(void)
+{
+    mbedtls_memory_buffer_alloc_init(_mbedtls_heap, sizeof(_mbedtls_heap));
+#if defined(CONFIG_CC310_BACKEND)
+    nrf_cc310_platform_abort_init();
+    return mbedtls_platform_setup(&platform_context);
+#endif
+    return 0;
+}
+
+/* Entries must correspond to order in enum test_vector_t found in common_test.h */
+const size_t test_vector_sizes[] = {
+	sizeof(test_vector_aes_t),
+	sizeof(test_vector_aead_t),
+	sizeof(test_vector_ecdsa_verify_t),
+	sizeof(test_vector_ecdsa_sign_t),
+	sizeof(test_vector_ecdsa_random_t),
+	sizeof(test_vector_ecdh_t),
+	sizeof(test_vector_hash_t),
+	sizeof(test_vector_hmac_t),
+	sizeof(test_vector_hkdf_t),
+	sizeof(test_vector_ecjpake_t),
+};
+
+/* Entries must correspond to order in enum test_vector_t found in common_test.h */
+const size_t test_vector_name_offset[] = {
+	offsetof(test_vector_aes_t, p_test_vector_name),
+	offsetof(test_vector_aead_t, p_test_vector_name),
+	offsetof(test_vector_ecdsa_verify_t, p_test_vector_name),
+	offsetof(test_vector_ecdsa_sign_t, p_test_vector_name),
+	offsetof(test_vector_ecdsa_random_t, p_test_vector_name),
+	offsetof(test_vector_ecdh_t, p_test_vector_name),
+	offsetof(test_vector_hash_t, p_test_vector_name),
+	offsetof(test_vector_hmac_t, p_test_vector_name),
+	offsetof(test_vector_hkdf_t, p_test_vector_name),
+	offsetof(test_vector_ecjpake_t, p_test_vector_name),
+};
+
+mbedtls_ctr_drbg_context ctr_drbg_ctx;
+
+static int entropy_func(void *ctx, unsigned char *buf, size_t len)
+{
+	return entropy_get_entropy(ctx, buf, len);
+}
+
+int init_ctr_drbg(const unsigned char *p_optional_seed, size_t len)
+{
+	static const unsigned char ncs_seed[] = "ncs_drbg_seed";
+
+	const unsigned char *p_seed;
+
+	if (p_optional_seed == NULL) {
+		p_seed = ncs_seed;
+		len = sizeof(ncs_seed);
+	} else {
+		p_seed = p_optional_seed;
+	}
+
+	struct device *p_device = device_get_binding(CONFIG_ENTROPY_NAME);
+	if (!p_device) {
+		return -ENODEV;
+	}
+
+	mbedtls_ctr_drbg_init(&ctr_drbg_ctx);
+	return mbedtls_ctr_drbg_seed(&ctr_drbg_ctx, entropy_func, p_device, p_seed, len);
+}
+
+uint32_t unhexify(uint8_t *p_output, char const *p_input)
+{
+	/* Assumes p_input is properly terminated and that the output buffer can accomodate the required size. */
+	size_t len = strlen(p_input);
+	return hex2bin(p_input, strlen(p_input), p_output, len);
```

Comment:
```suggestion
	return hex2bin(p_input, len, p_output, len);
```

Reply:
Changed to use hex2bin directly everywhere in 048b58e